### PR TITLE
Introduce Waypoint classes for describing targets with multiple points.

### PR DESCRIPTION
**Pull request by yck011522**
This PR introduces a new class similar to Targets that are used for `plan_motion()`, but now for a sequence of targets. This is used for planning motions with multiple segments similar to the current `plan_cartesian_motion()`.

The intended structure is a "Waypoints" parent class with two childs "FrameWaypoints" and "PointAxisWaypoints". First is used with a list of Frames similar to the current plan_cartesian_motion use case, second is used for 3D printing type of work.

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas_fab.robots.CollisionMesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by jf---** [comment]
`PointAxisWaypoints` is exciting!

**Comment by yck011522** [comment]
@gonzalocasas ping

File: src/compas_fab/robots/targets.py [hunk 1]
```diff
@@ -119,6 +124,7 @@ def __data__(self):
             "tolerance_position": self.tolerance_position,
             "tolerance_orientation": self.tolerance_orientation,
             "tool_coordinate_frame": self.tool_coordinate_frame,
+            "name": self.name,
```
Comment (yck011522) [inline src/compas_fab/robots/targets.py:127]: This was a mistake having name left out in previous PR

File: src/compas_fab/robots/targets.py [hunk 2]
```diff
@@ -221,7 +230,9 @@ class PointAxisTarget(Target):
         The tool tip coordinate frame relative to the flange coordinate frame of the robot.
         If not specified, the target point is relative to the robot's flange (T0CF) and the
         Z axis of the flange can rotate around the target axis.
-
+    nane : str, optional
```
Comment (yck011522) [inline src/compas_fab/robots/targets.py:233]:
```suggestion
    name : str, optional
```

File: src/compas_fab/robots/targets.py [hunk 3]
```diff
@@ -261,9 +274,9 @@ def scaled(self, factor):
         """
         target_point = self.target_point.scaled(factor)
         tolerance_position = self.tolerance_position * factor if self.tolerance_position else None
-        target_z_axis = self.target_z_axis.scaled  # Vector is unitized and is not scaled
+        target_z_axis = self.target_z_axis  # Vector is unitized and is not scaled
```
Comment (yck011522) [inline src/compas_fab/robots/targets.py:277]: Mistake in the past

File: src/compas_fab/robots/targets.py [hunk 4]
```diff
@@ -261,9 +274,9 @@ def scaled(self, factor):
...
         tool_coordinate_frame = self.tool_coordinate_frame.scaled(factor) if self.tool_coordinate_frame else None
-        return PointAxisTarget(target_point, target_z_axis, tool_coordinate_frame, tolerance_position, self.name)
+        return PointAxisTarget(target_point, target_z_axis, tolerance_position, tool_coordinate_frame, self.name)
```
Comment (yck011522) [inline src/compas_fab/robots/targets.py:279]: Also a stupid mistake in the past PR. I didnt have testing back then.

File: tests/robots/test_targets.py [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+        target_configuration, tolerance_above, tolerance_below
+    )
+    target = ConstraintSetTarget(constraint_set, name)
+    nt = ConstraintSetTarget.__from_data__(target.__data__)
+    assert target.constraint_set == nt.constraint_set
+    assert target.name == nt.name
+    for c1, c2 in zip(target.constraint_set, nt.constraint_set):
+        assert c1.__data__ == c2.__data__
+
+    # ConstraintSetTarget with OrientationConstraint and Po
```
Comment (yck011522) [inline tests/robots/test_targets.py:105]:
```suggestion
    # ConstraintSetTarget with OrientationConstraint and PositionConstraint
```

File: src/compas_fab/robots/targets.py [hunk 6]
```diff
@@ -451,3 +476,243 @@ def scaled(self, factor):
...
+            The scaling factor.
+
+        Returns
+        -------
+        :class:`FrameWaypoints`
+            The scaled frame waypoints.
+        """
+        target_frames = [frame.scaled(factor) for frame in self.target_frames]
+        tolerance_position = self.tolerance_position * factor
+        tolerance_orientation = self.tolerance_orientation * factor
```
Comment (gonzalocasas) [inline src/compas_fab/robots/targets.py:643]:
```suggestion
        tolerance_orientation = self.tolerance_orientation
```

**Review by gonzalocasas — APPROVED** [review]
LGTM! 🚀

File: tests/robots/test_targets.py [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+    assert point_axis_waypoints.tool_coordinate_frame == nt.tool_coordinate_frame
+    assert point_axis_waypoints.name == nt.name
+
+
+def test_target_scale(frame_target):
+    scale_factor = 0.001
+    nt = frame_target.scaled(scale_factor)
+    assert nt.target_frame == frame_target.target_frame.scaled(scale_factor)
+    assert nt.tolerance_position == frame_target.tolerance_position * scale_factor
+    assert nt.tolerance_orientation == frame_target.tolerance_orientation * scale_factor
```
Comment (gonzalocasas) [inline tests/robots/test_targets.py:175]:
```suggestion
    # orientation doesn't need scale
    assert nt.tolerance_orientation == frame_target.tolerance_orientation
```

File: tests/robots/test_targets.py [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+            )
+        else:
+            assert nt.target_configuration.joint_values[i] == configuration_target.target_configuration.joint_values[i]
+
+
+def test_frame_waypoints_scale(frame_waypoints):
+    scale_factor = 0.001
+    nt = frame_waypoints.scaled(scale_factor)
+    assert nt.tolerance_position == frame_waypoints.tolerance_position * scale_factor
+    assert nt.tolerance_orientation == frame_waypoints.tolerance_orientation * scale_factor
```
Comment (gonzalocasas) [inline tests/robots/test_targets.py:212]:
```suggestion
    # orientation doesn't need scale
    assert nt.tolerance_orientation == frame_waypoints.tolerance_orientation
```

File: src/compas_fab/robots/targets.py [hunk 9]
```diff
@@ -451,3 +477,243 @@ def scaled(self, factor):
...
+
+    Waypoints represent a sequence of targets the robot should pass through in the order they are defined.
+    This is in contrast to :class:`Target` which represent only a single target.
+    The initial (starting) point should not be included in the waypoints list.
+    It is valid for a Waypoints object to have one target.
+
+    Waypoints are useful for tasks like painting, welding, or 3D printing, where the programmer
+    wants to define the waypoints the robot should pass through.
+
+    Waypoints are intended to be used for motion planning with a planning backend by using :meth:`compas_fab.robot.plan_motion_with_waypoints`.
```
Comment (gonzalocasas) [inline src/compas_fab/robots/targets.py:493]:
```suggestion
    Waypoints are intended to be used for motion planning with a planning backend by using :meth:`compas_fab.robot.plan_cartesian_motion`.
```

**Pull request merged**